core/src/executor/evaluate/evaluated.rs

```diff
@@ -210,22 +341,469 @@ impl<'a> Evaluated<'a> {
+                })
+            }
+            Some(TrimWhereField::Leading) => {
+                let pivot = matched_vec
+                    .iter()
+                    .enumerate()
+                    .skip_while(|(vec_idx, (slice_idx, _))| vec_idx == slice_idx)
+                    .map(|(vec_idx, (_, _))| vec_idx)
+                    .next();
+
+                let start = match pivot {
+                    Some(idx) => match idx {
+                        0 => 0,
+                        _ => matched_vec[idx - 1].0 + 1,
+                    },
+                    _ => matched_vec[matched_vec.len() - 1].0 + 1,
+                };
+
+                Ok(Evaluated::StrSlice {
+                    source: expr_str.to_owned(),
+                    range: range.start + start..range.end,
+                })
+            }
+            Some(TrimWhereField::Trailing) => {
+                let pivot = matched_vec
+                    .iter()
+                    .rev()
+                    .enumerate()
+                    .skip_while(|(vec_idx, (slice_idx, _))| {
+                        *vec_idx == sliced_expr.len() - slice_idx - 1
+                    })
+                    .map(|(vec_idx, (_, _))| vec_idx)
+                    .next();
+
+                let end = match pivot {
+                    Some(idx) => match idx {
+                        0 => range.end,
+                        _ => matched_vec[matched_vec.len() - idx].0,
+                    },
+                    _ => matched_vec[0].0,
+                };
+
+                Ok(Evaluated::StrSlice {
+                    source: expr_str.to_owned(),
+                    range: range.start..end,
+                })
+            }
+            None => {
+                let start = expr_str
+                    .chars()
+                    .skip(range.start)
+                    .enumerate()
+                    .find(|(_, c)| !c.is_whitespace())
+                    .map(|(idx, _)| idx + range.start)
+                    .unwrap_or(0);
+
+                let end = expr_str.len()
+                    - expr_str
+                        .chars()
+                        .rev()
+                        .skip(expr_str.len() - range.end)
+                        .enumerate()
+                        .find(|(_, c)| !c.is_whitespace())
+                        .map(|(idx, _)| expr_str.len() - (range.end - idx))
+                        .unwrap_or(0);
+
+                Ok(Evaluated::StrSlice {
+                    source,
+                    range: start..end,
+                })
+            }
+        }
+    }
+
     pub fn try_into_value(self, data_type: &DataType, nullable: bool) -> Result<Value> {
         let value = match self {
-            Evaluated::Value(v) => v,
             Evaluated::Literal(v) => Value::try_from_literal(data_type, &v)?,
+            Evaluated::Value(v) => v,
+            Evaluated::StrSlice {
```

Comment:
I would like to know how to write a test here

Reply:
I found the use of the function in insert.rs and filled in the test.
It was possible in the form of 
```sql
INSERT INTO a VALUES (SUBSTR(b)) 
```